### PR TITLE
Add support for Django 4.2 and 5.0, and drop support for prior versions

### PR DESCRIPTION
**Pull request by pamella**
## Description

- Add support for Django 4.2 and 5.0, and drop support for prior versions.
  - https://www.djangoproject.com/download/
image
  - https://docs.djangoproject.com/en/5.0/faq/install/#what-python-version-can-i-use-with-django
image
- Replace Travis with GitHub Actions.
- Replace `codecov` with `coveralls`.
  - https://coveralls.io/github/vintasoftware/django-celerybeat-status 
- Refactor code to reflect newer versions of Python and Django.
- Run `black` and `isort`.
- Add badges to README.
- -Add Contributing section to README.

:information_source: Related to #23.

## Related issues

- Closes #20 
- Closes #22 
- Closes #23

## Screenshots 

- Run the project.
- Check if the admin pages are working properly.

[Screencast from 16-05-2024 19:16:15.webm](https://github.com/vintasoftware/django-celerybeat-status/assets/22326737/962ec81c-a48b-49ab-917b-008cce027de0)

### CODE REVIEW

File: celerybeat_status/__init__.py [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # -*- coding: utf-8
-__title__ = 'CeleryBeat Status'
-__version__ = '0.0.10'
-__author__ = 'Vinta Software'
-__license__ = 'MIT License'
-__copyright__ = 'Copyright 2017 Vinta Serviços e Soluções Tecnológicas Ltda'
+__title__ = "CeleryBeat Status"
+__version__ = "0.1.0"
```
Comment (pamella) [inline celerybeat_status/__init__.py]: Let me know if this new version makes sense, considering that this PR adds breaking changes.

Comment (fjsj) [reply celerybeat_status/__init__.py]: I guess we can jump directly to 1.

File: manage.py [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
+            "available on your PYTHONPATH environment variable? Did you "
+            "forget to activate a virtual environment?"
+        ) from exc
+    execute_from_command_line(sys.argv)
 
-    execute_from_command_line(sys.argv)
\ No newline at end of file
+
+if __name__ == "__main__":
+    main()
```
Comment (fjsj) [inline manage.py:22]: Do we need to keep this manage.py?

Comment (fjsj) [reply manage.py:22]: In django-virtual-models, I have a separate example folder: https://github.com/vintasoftware/django-virtual-models

**Review by fjsj — APPROVED** [review]
Overall LGTM. What's missing?

**Comment by pamella** [comment]
> Overall LGTM. What's missing?

The TODO items from the description:
- Add GitHub action to address the release process. 
- Update README with the new Django syntax and instructions on how to run/contribute to this project.

File: manage.py (hunk 2 above)
Comment (pamella) [reply manage.py:22]: I'll check

Comment (pamella) [reply manage.py:22]: I believe folks did it like that because of the need to add `celerybeat_status` to `INSTALLED_APPS`.

Comment (pamella) [reply manage.py:22]: Would renaming the `tests` folder to `example` be a middle ground?

Comment (fjsj) [reply manage.py:22]: yes, please do

Comment (fjsj) [reply manage.py:22]: But I don't think Django Apps need manage.py. I see this project has one, but I think it's to run tests: https://github.com/django-extensions/django-extensions

Comment (pamella) [reply manage.py:22]: Yes, it's for testing and running the example

Comment (pamella) [reply manage.py:22]: Considering the [django-extensions](https://github.com/django-extensions/django-extensions), maybe having the test folder is a pattern?

Comment (pamella) [reply manage.py:22]: As for the `tests` folder, it seems to be a pattern indeed, I think we should keep it as it is in this case.

Other django apps:

- https://github.com/encode/django-rest-framework
- https://github.com/adamchainz/django-cors-headers
- https://github.com/graphql-python/graphene-django

Comment (pamella) [reply manage.py:22]: I'm trying to find an example with the `manage.py`, but so far I believe we have it so we can run and render the admin UI for testing/QA purposes.

**Comment by pamella** [comment]
@fjsj The PR is ready for review. I will add the release GitHub action in another PR.

File: manage.py (hunk 2 above)
Comment (fjsj) [reply manage.py:22]: Having a tests folder is fine, but that's where the tests are, not the examples.
The idea for having a tests dir is to not distribute it along with the main app (it's not installed via pip for example)

Comment (pamella) [reply manage.py:22]: I see, but the `tests` folders have auxiliary files besides the `test_*` ones. For example:
-  `django-rest-framework` has [`models.py`](https://github.com/encode/django-rest-framework/blob/master/tests/models.py) and [`urls.py`](https://github.com/encode/django-rest-framework/blob/master/tests/urls.py).
- `django-cors-headers` has [`settings.py`](https://github.com/adamchainz/django-cors-headers/blob/main/tests/settings.py) and [`urls.py`](https://github.com/adamchainz/django-cors-headers/blob/main/tests/urls.py).

We would be doing something similar here

Comment (fjsj) [reply manage.py:22]: yes, that's fine

Comment (fjsj) [reply manage.py:22]: Django own code also does this.

**Pull request merged**